modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala

```diff
@@ -240,6 +241,10 @@ class Engine[F[_]: MonadError[?[_], Throwable]: Logger, S, U](stateL: Engine.Sta
       _.current.execution.forall(Action.completed).option(Handle.fromStream[F, S, EventType](Stream(executed(id))))
     ).getOrElse(unit))
 
+  private def abort[R <: RetVal](id: Observation.Id, i: Int, r: Result.OKAborted[R]): HandleType[Unit] =
+    modifyS(id)(_.mark(i)(r)) *>
```

Comment:
It seems to me that here we should set the `StepState` to be `Aborted` but I don't know how